src/main/java/core/basesyntax/service/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(Map<String, Integer> inventory, FruitTransaction transaction) {
+        inventory.merge(transaction.getFruit(), transaction.getQuantity(), Integer::sum);
```

Comment:
The use of `Integer::sum` is incorrect for purchase operations. You should subtract the transaction quantity from the current inventory instead of adding it. Consider using a lambda expression to perform the subtraction.